src/client.h

```diff
@@ -585,9 +600,24 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 	//s32 m_daynight_i;
 	//u32 m_daynight_ratio;
 	std::queue<std::wstring> m_chat_queue;
+
+	// The authentication methods we can use to enter sudo mode (=change password)
+	u32 m_sudo_auth_methods;
+
 	// The seed returned by the server in TOCLIENT_INIT is stored here
 	u64 m_map_seed;
+
+	// Auth data
+	std::string m_playername;
 	std::string m_password;
+	// If set, this will be sent (and cleared) upon a TOCLIENT_ACCEPT_SUDO_MODE
+	std::string m_new_password;
+	// Usable by auth mechanisms.
+	int m_authstate;
+	int m_chosen_auth_mech;
+	void * m_auth_data;
```

Comment:
I would've personally implemented this a bit differently.  Here we have a decent case for making AuthMechanism its own virtual base class (read: interface) with each auth mechanism an implementation of this.  It would store the SRPData or whatever inside of this object.